lang/test/org/partiql/lang/syntax/SqlParserDateTimeTests.kt

```diff
@@ -25,6 +34,567 @@ class SqlParserDateTimeTests : SqlParserTestBase() {
                 project = projectList(projectExpr(date(2021, 3, 10))),
                 from = scan(id("foo"))
             )
+        },
+        DateTimeTestCase("TIME '02:30:59'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME (3) '12:59:31'") {
+            time(12, 59, 31, 0, 3, null)
+        },
+        DateTimeTestCase("TIME '23:59:59.9999'") {
+            time(23, 59, 59, 999900000, 9, null)
+        },
+        DateTimeTestCase("TIME (7) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 7, null)
+        },
+        DateTimeTestCase("TIME (9) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
+        DateTimeTestCase("TIME (0) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 0, null)
+        },
+        DateTimeTestCase("TIME (10) '23:59:59.123456789'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
+        DateTimeTestCase("TIME '02:30:59-05:30'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME '02:30:59+05:30'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME '02:30:59-14:39'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME '02:30:59+00:00'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME '02:30:59-00:00'") {
+            time(2, 30, 59, 0, 9, null)
+        },
+        DateTimeTestCase("TIME (3) '12:59:31+10:30'") {
+            time(12, 59, 31, 0, 3, null)
+        },
+        DateTimeTestCase("TIME (0) '00:00:00+00:00'") {
+            time(0, 0, 0, 0, 0, null)
+        },
+        DateTimeTestCase("TIME (0) '00:00:00-00:00'") {
+            time(0, 0, 0, 0, 0, null)
+        },
+        DateTimeTestCase("TIME '23:59:59.9999-11:59'") {
+            time(23, 59, 59, 999900000, 9, null)
+        },
+        DateTimeTestCase("TIME (7) '23:59:59.123456789+01:00'") {
+            time(23, 59, 59, 123456789, 7, null)
+        },
+        DateTimeTestCase("TIME (9) '23:59:59.123456789-14:50'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
+        DateTimeTestCase("TIME (0) '23:59:59.123456789-18:00'") {
+            time(23, 59, 59, 123456789, 0, null)
+        },
+        DateTimeTestCase("TIME (10) '23:59:59.123456789+18:00'") {
+            time(23, 59, 59, 123456789, 9, null)
+        },
```

Comment:
Ack, the precision specified by the user is substituted with the maximum precision supported?  Should be an error.

Reply:
The Postgres I verified with uses the default precision as the maximum precision. It also uses the maximum precision if the specified precision is more than the maximum.

Reply:
yes, yes, not sure where I got the idea that it was an error from...